proxy.go

```diff
@@ -158,17 +158,17 @@ func (p *proxy) Destroy() error {
 	return nil
 }
 
-func (p *proxy) validateAndSerialize(arg1 interface{}) (serialization.Data, error) {
+func (p *proxy) validateAndSerialize(arg1 interface{}) (*iserialization.Data, error) {
 	if nilutil.IsNil(arg1) {
-		return nil, hzerror.NewHazelcastNilPointerError(bufutil.NilArgIsNotAllowed, nil)
+		return nil, hzerror.NewHazelcastNilPointerError("nil arg is not allowed", nil)
```

Comment:
nit: should probably be:
```suggestion
		return nil, errNilArg
```

Reply:
Fixed at https://github.com/hazelcast/hazelcast-go-client/pull/555/commits/f8335e3a66165c8383ce93ed08fc6127f3af1375